🎨 [New Terms]: Java Math Methods: `logarithmic functions` (#902) (#1181)
Codecademy/docs: content/java/concepts/math-methods/terms/log10/log10.md

---

> Title: '.log10()'

Reviewer: ```suggestion
'Math.log10()'
```

Description: 'Returns the logarithm base 10 of a double value as a parameter.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Arithmetic'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

The **`.log10()`** method returns `log a`, the logarithm of `a` with base 10.

## Syntax

```pseudo
Math.log10(a);
```

- The `.log10()` method takes one parameter `a`.
- If `a` is `NaN`, then return value is `NaN`.
- If `a` is `negative number`, then return value is `NaN`.
- If `a` is `positive number`, then return value is `double`.
- If `a` is `positive / negative zero`, then return value is `positive infinity`.

> - If `a` is `positive infinity`, then return value is `positive infinity`.

Reviewer: ```suggestion
If `a` is positive infinity, the return value is positive infinity.
```

## Example

> The following example demonstrates using `.log10()` to find logs (with base 10):

Reviewer: ```suggestion
The following example demonstrates using `Math.log10()` to find base-10 logarithms:
```

```java
import java.lang.Math;

public class Main {
    public static void main(String args[]) {
        double a = 10;
        double b = -6;
        double c = 0;

        System.out.println("The result of log10(" + a + ") is: " + Math.log10(a));
        System.out.println("The result of log10(" + b + ") is: " + Math.log10(b));
        System.out.println("The result of log10(" + c + ") is: " + Math.log10(c));
    }
}
```

This will produce the following output:

```shell

> The result of log10(10) is: 1

Reviewer 2: ```suggestion
The result of log10(10.0) is: 1.0
```

> The result of log10(-6) is: NaN

Reviewer 2: ```suggestion
The result of log10(-6.0) is: NaN
```

> The result of log10(0) is: -Infinity

Reviewer 2: ```suggestion
The result of log10(0.0) is: -Infinity
```
